### PR TITLE
[BUGFIX] Snowflake/Oracle/DB2 <--> SQLAlchemy table and column names case insensitivity representation

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
Case-insensitivity capture for  Snowflake/Oracle/DB2 with respect to SQLAlchemy is made more robust (tests updated).

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], or [CONTRIB]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: DEVREL-2510
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in GitHub issues or Slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/expectations/metrics/util.py [hunk 1]
```diff
@@ -666,73 +662,101 @@ def get_dbms_compatible_column_names(
...
+        return typed_batch_column_names_list
 
-    if is_list:
-        return typed_column_names_list
-
-    return typed_column_names_list[0]
+    return typed_batch_column_names_list[0]
 
 
 def verify_column_names_exist(
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/metrics/util.py:691]: Love it.

File: great_expectations/expectations/metrics/map_metric_provider.py [hunk 2]
```diff
@@ -2326,7 +2298,7 @@ def _sqlalchemy_column_pair_map_condition_filtered_row_count(
     column_B_name = accessor_domain_kwargs["column_B"]
 
     column_names: List[Union[str, quoted_name]] = [column_A_name, column_B_name]
-    verify_column_names_exist(
+    _ = verify_column_names_exist(
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/metrics/map_metric_provider.py]: I though `verify_column_names_exist()` returned `None`. Is there a reason for the `_`?

Comment (Shinnnyshinshin) [reply great_expectations/expectations/metrics/map_metric_provider.py]: thought*

Comment (alexsherstinsky) [reply great_expectations/expectations/metrics/map_metric_provider.py]: @Shinnnyshinshin Brilliant catch!  Fixing (in next commit).  Thank you!

**Review by Shinnnyshinshin — APPROVED** [review]
🚀

File: great_expectations/expectations/metrics/util.py [hunk 3]
```diff
@@ -666,73 +662,101 @@ def get_dbms_compatible_column_names(
...
 
     Returns:
         Single property-typed column name object or list of property-typed column name objects (depending on input).
     """
-    verify_column_names_exist(
-        column_names=column_names,
-        batch_columns_list=batch_columns_list,
-        error_message_template=error_message_template,
+    normalized_typed_batch_columns_mappings: List[
+        Tuple[str, str | sqlalchemy.sql.quoted_name]
```
Comment (cdkini) [inline great_expectations/expectations/metrics/util.py:671]: nit - we don't need `List` and `Tuple` we can just use `list` and `tuple`

File: great_expectations/expectations/metrics/util.py [hunk 4]
```diff
@@ -666,73 +662,101 @@ def get_dbms_compatible_column_names(
...
 ) -> None:
+    _ = verify_column_names_exist_and_get_normalized_typed_column_names_map(
+        column_names=column_names,
+        batch_columns_list=batch_columns_list,
+        error_message_template=error_message_template,
+        verify_only=True,
+    )
+
+
+def verify_column_names_exist_and_get_normalized_typed_column_names_map(
```
Comment (cdkini) [inline great_expectations/expectations/metrics/util.py]: Is this used externally outside of this file? If not, let's prefix with `_` to denote that it is private

File: great_expectations/expectations/metrics/util.py (hunk 3 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/metrics/util.py:671]: @cdkini Without `List` and `Tuple`, how can we specify what the internal types are?..  Thanks.

File: great_expectations/expectations/metrics/util.py (hunk 4 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/metrics/util.py]: @cdkini Great idea -- done (committed).  Thanks.

**Pull request merged**